### PR TITLE
[Bugfix] Moar descriptive error messages

### PR DESCRIPTION
**Pull request by lkudyba**
@adamzima @mhib @jtkpiotr

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -9,13 +9,23 @@ class Event < ApplicationRecord
...
   def start_time_must_be_lower_than_end_time
-    errors.add(:start_time, "Can't be higher than end_time") if start_time.try(:>=, end_time)
+    errors.add(:start_time, "Start time must be lower than end time") if start_time.try(:>=, end_time)
   end
 
   def no_collision
     return unless start_time && end_time
-    return unless Event.in_span_for_conference_room(start_time, end_time, conference_room).exists?
-    errors.add(:start_time, "Can't start while another event is in progress")
+    event = Event.in_span_for_conference_room(start_time, end_time, conference_room).first
```
Comment (pbzymek) [inline app/models/event.rb]: would it make sense to do:
`conference_room.events.in_span(start_time, end_time)` ?

File: app/models/event.rb [hunk 2]
```diff
@@ -9,23 +9,29 @@ class Event < ApplicationRecord
   validate :no_collision
 
   def start_time_must_be_lower_than_end_time
-    errors.add(:start_time, "Can't be higher than end_time") if start_time.try(:>=, end_time)
+    errors.add(:start_time, "Start time must be lower than end time") if start_time.try(:>=, end_time)
```
Comment (adamzima) [inline app/models/event.rb]: Nice, now we can use single quotes :D

Comment (adamzima) [inline app/models/event.rb]: How about adding `return unless start_time && end_time` rather than using `#try`

File: app/models/event.rb [hunk 3]
```diff
@@ -9,23 +9,29 @@ class Event < ApplicationRecord
...
   end
 
   def no_collision
     return unless start_time && end_time
-    return unless Event.in_span_for_conference_room(start_time, end_time, conference_room).exists?
-    errors.add(:start_time, "Can't start while another event is in progress")
+    event = conference_room.events.in_span(start_time, end_time).first
+    return unless event
+
+    event_in_progress_text = "Another event already in progress in #{conference_room.title} (#{event.start_time.strftime('%H:%M')} - #{event.end_time.strftime('%H:%M')})"
```
Comment (adamzima) [inline app/models/event.rb]: That's a looong line - can we split it?

File: app/assets/javascripts/calendar/create_event.js [hunk 4]
```diff
@@ -1,7 +1,17 @@
 $(function () {
     $(document).on('ajax:success', '#create-event-form', function () {
         $('#create-event-modal').modal('hide');
-    }).on('ajax:error', '#create-event-form', function (e, xhr, status, err) {
-        $('#create-event-error').text('Failed to create event: ' + err).show();
-    })
+    }).on('ajax:error', '#create-event-form', function (e, xhr) {
+        var responseObj = JSON.parse(xhr.responseText);
+
+        $('.form-group').removeClass('has-error')
```
Comment (s1mplex) [inline app/assets/javascripts/calendar/create_event.js]: When chaining prefer one method per line, eg:

`
$('.form-group').
  removeClass('has-error').
  children('.error-text').
  empty();
`

Comment (adamzima) [inline app/assets/javascripts/calendar/create_event.js]: When commenting use formatting :P

```js
$('.form-group').
    removeClass('has-error').
    children('.error-text').
    empty();
```

File: app/models/event.rb [hunk 5]
```diff
@@ -9,23 +9,30 @@ class Event < ApplicationRecord
   validate :no_collision
 
   def start_time_must_be_lower_than_end_time
-    errors.add(:start_time, "Can't be higher than end_time") if start_time.try(:>=, end_time)
+    return unless start_time && end_time
```
Comment (s1mplex) [inline app/models/event.rb]: Do you have to check for `start_time` and `end_time` when you already validate their presence?

File: app/models/event.rb [hunk 6]
```diff
@@ -9,23 +9,30 @@ class Event < ApplicationRecord
...
+    errors.add(:start_time, 'Start time must be lower than end time')
   end
 
   def no_collision
     return unless start_time && end_time
```
Comment (s1mplex) [inline app/models/event.rb]: Same here

File: app/models/event.rb [hunk 7]
```diff
@@ -9,23 +9,30 @@ class Event < ApplicationRecord
...
+    if start_time < event.end_time && start_time >= event.start_time
+      errors.add(:start_time, event_in_progress_text)
+    end
+
+    if end_time > event.start_time && end_time <= event.end_time
+      errors.add(:end_time, event_in_progress_text)
+    end
   end
 
   scope :in_span, -> (starting, ending) {
```
Comment (s1mplex) [inline app/models/event.rb]: Prefer having class macros like `scope`, `attr_accessor`, etc. higher than method definitions

File: app/models/event.rb (hunk 5 above)
Comment (lkudyba) [inline app/models/event.rb]: Yes

File: app/models/event.rb [hunk 8]
```diff
@@ -34,4 +20,26 @@ def no_collision
...
+    return unless start_time >= end_time
+    errors.add(:start_time, 'Start time must be lower than end time')
+  end
+
+  def no_collision
+    return unless start_time && end_time
+    events = conference_room.events.in_span(start_time, end_time)
+    return unless events.exists?
+
+    event_in_progress_text = lambda do |event|
```
Comment (adamzima) [inline app/models/event.rb]: Why not new lambda syntax `->`?

**Pull request merged**